### PR TITLE
Add Python package

### PR DESCRIPTION
**Pull request by nokome**
This adds a setup for turning the Python language bindings into a Python package so that we can distribute them on PyPI.

I have added `toJSON` and `fromJSON` so that we have an integration point with other tools e.g. 

```bash
python3 article.py | npx encoda - --from json article.docx
```

I think that is as much functionality that we want to add to the Python package beyond the `types.py`. Conversion to/from Python types e.g. pandas data frame will go in the `stencila/py` package that will depend upon this package.

@beneboy Could you take this on. I have just quickly copied over files from a Python package I worked on at the end of the year. Linting fails and the test coverage is low. See the `*-py` tasks in the `Makefile` and the new `.travis.yaml` file. We'll aim to release a v1.0-beta version of the python package along side those for R and Typescript at the end of the month.

### CODE REVIEW

**Comment by nokome** [comment]
I'm have second thoughts about this and #103. Those thoughts fall into the "Where should we put this code?" category that we often come across. The original motivation for creating a `stencila_schema` Python package was:

- to provide a way for https://github.com/stencila/py to depend upon the language bindings generated here
- to provide a package for other people to develop Python tooling around the schema, but not rely on the https://github.com/stencila/py (which may end up having a dependencies like Pandas)

However, the problems with this are:

- yet another package for us to maintain (Python linting, PyPI deployment etc)
- making this packish a bit of a multilanguage Frankenstein (not a big issue but can be confusing to see a `setup.py` and a `package.json` and a `DESCRIPTION` and ... in the same repo)

I'm more concerned with the first problem. I suggest that rather than creating a `stencila_schema` Python package that we use git submodules to depend upon this repo in https://github.com/stencila/py. Others could do the same if they didn't want to use `stencila/py`.

If we go ahead we would move the `python/util.py` file to `stencila/py` but still generate Python bindings - in the form of a `types.py` that would be committed.

All this holds for the R package in #103.

**Comment by beneboy** [comment]
Could the schema repo contain the required `setup.py` for uploading to PyPI in a subdirectory? Then we just generate the Python files inside there. Going up to PyPI is pretty simple.

**Comment by nokome** [comment]
Yes, we could do that.

My main concern is the maintenance burden, rather than repo structure, but
maybe it's not a big deal. I think we already have most of the setup we
need e.g lint and test setup and a setup.py. And there is probably not
going to be changes needed if we keep the bindings light weight.

And maybe depending upon a PyPI package, is easier than using Git
submodules.

As you can see, I'm ambivalent as to which way to go. Although, I think you
are swaying me back to continuing this PR and have a package here.

**Comment by nokome** [comment]
@beneboy After doing more work on the R package in https://github.com/stencila/schema/pull/103 my thinking is that we should go back to the structure we had back in December with a separate folder for each language: https://github.com/stencila/schema/tree/0a731f60ce2736947f68c200d430258d3339d73a

You should be able to start by moving the `setup.py`, `MANIFEST.ini, `tests/*_test.py` etc into the `py` directory.

**Comment by nokome** [comment]
I've created a `py/Makefile` and used it in `.travis.yml` to mirror the approach in #103. Also temporarily removed the Node.js build from `.travis.yml`.

File: py/stencila/schema/util.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+def to_dict(node: Entity) -> dict:
+    """Convert an Entity node to a dictionary"""
+    node_dict = {
+        "type": node.__class__.__name__
+    }
+    node_dict.update(node.__dict__)
+    return node_dict
+
+
+def from_dict(node_dict: dict) -> typing.Union[dict, Node]:
```
Comment (nokome) [inline py/stencila/schema/util.py]: This does not handle child entities. It should traverse all descendant nodes and convert them too. See how we do it for R (where a `list` is the similar to a Python `dict`):

https://github.com/stencila/schema/blob/589adcff6b3c5db8c86a9c4a09b46db12f35278c/r/R/entity.R#L30-L31

Generally, we might save ourselves time by structuring the code similarly across the packages for the different languages.

**Comment by nokome** [comment]
I've created issue #106 but have attached it to https://github.com/stencila/schema/milestone/2 and so it's a lower priority than the issues in https://github.com/stencila/schema/milestone/1. 

Let's get this PR merged soon so we can focus on getting the docs fixed up and on Encoda. I think this just needs higher test coverage before we do that.

**Pull request merged**